Getting io.grpc.StatusRuntimeException

Hi,

I am getting following error while transcribing long audio:
                                                                           

> Error calling the API.
>  io.grpc.StatusRuntimeException: UNKNOWN
>                                                                                 at io.grpc.Status.asRuntimeException(Status.java:545)
>                                                                                 at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:370)
>                                                                                 at io.grpc.ClientInterceptors$CheckedForwardingClientCall.start(ClientInterceptors.java:203)
>                                                                                 at io.grpc.stub.ClientCalls.startCall(ClientCalls.java:248)
>                                                                                 at io.grpc.stub.ClientCalls.asyncStreamingRequestCall(ClientCalls.java:241)
>                                                                                 at io.grpc.stub.ClientCalls.asyncBidiStreamingCall(ClientCalls.java:106)
>                                                                                 at com.google.cloud.speech.v1beta1.SpeechGrpc$SpeechStub.streamingRecognize(SpeechGrpc.java:232)
>                                                                                 at com.google.cloud.android.speech.ApiFragment.startRecognizing(ApiFragment.java:188)
>                                                                                 at com.google.cloud.android.speech.MainActivity$1.onVoiceStart(MainActivity.java:64)
>                                                                                 at com.google.cloud.android.speech.VoiceRecorder$ProcessVoice.run(VoiceRecorder.java:201)
>                                                                                 at java.lang.Thread.run(Thread.java:818)
>                                                                              Caused by: java.lang.IllegalStateException: OAuth2Credentials instance does not support refreshing the access token. An instance with a new access token should be used, or a derived type that supports refreshing should be used.
>                                                                                 at com.google.auth.oauth2.OAuth2Credentials.refreshAccessToken(OAuth2Credentials.java:115)
>                                                                                 at com.google.auth.oauth2.OAuth2Credentials.refresh(OAuth2Credentials.java:97)
>                                                                                 at com.google.auth.oauth2.OAuth2Credentials.getRequestMetadata(OAuth2Credentials.java:74)
>                                                                                 at com.google.cloud.android.speech.ApiFragment$GoogleCredentialsInterceptor.getRequestMetadata(ApiFragment.java:326)
>                                                                                 at com.google.cloud.android.speech.ApiFragment$GoogleCredentialsInterceptor.access$200(ApiFragment.java:244)
>                                                                                 at com.google.cloud.android.speech.ApiFragment$GoogleCredentialsInterceptor$1.checkedStart(ApiFragment.java:268)
>                                                                                 at io.grpc.ClientInterceptors$CheckedForwardingClientCall.start(ClientInterceptors.java:195)
>                                                                                 at io.grpc.stub.ClientCalls.startCall(ClientCalls.java:248) 
>                                                                                 at io.grpc.stub.ClientCalls.asyncStreamingRequestCall(ClientCalls.java:241) 
>                                                                                 at io.grpc.stub.ClientCalls.asyncBidiStreamingCall(ClientCalls.java:106) 
>                                                                                 at com.google.cloud.speech.v1beta1.SpeechGrpc$SpeechStub.streamingRecognize(SpeechGrpc.java:232) 
>                                                                                 at com.google.cloud.android.speech.ApiFragment.startRecognizing(ApiFragment.java:188) 
>                                                                                 at com.google.cloud.android.speech.MainActivity$1.onVoiceStart(MainActivity.java:64) 
>                                                                                 at com.google.cloud.android.speech.VoiceRecorder$ProcessVoice.run(VoiceRecorder.java:201) 
>                                                                                 at java.lang.Thread.run(Thread.java:818) 

I cannot find out the reason for this issue.

--- a/speech/Speech/.gitignore
+++ b/speech/Speech/.gitignore
@@ -1,8 +1,7 @@
 *.iml
 .gradle
 /local.properties
-/.idea/workspace.xml
-/.idea/libraries
+/.idea
 .DS_Store
 /build
 /captures

--- a/speech/Speech/app/src/main/java/com/google/cloud/android/speech/VoiceRecorder.java
+++ b/speech/Speech/app/src/main/java/com/google/cloud/android/speech/VoiceRecorder.java
@@ -39,6 +39,7 @@ public class VoiceRecorder {
 
     private static final int AMPLITUDE_THRESHOLD = 1500;
     private static final int SPEECH_TIMEOUT_MILLIS = 2000;
+    private static final int MAX_SPEECH_LENGTH_MILLIS = 30 * 1000;
 
     public static abstract class Callback {
 
@@ -76,6 +77,9 @@ public class VoiceRecorder {
 
     /** The timestamp of the last time that voice is heard. */
     private long mLastVoiceHeardMillis = Long.MAX_VALUE;
+
+    /** The timestamp when the current voice is started. */
+    private long mVoiceStartedMillis;
 
     public VoiceRecorder(@NonNull Callback callback) {
         mCallback = callback;
@@ -180,22 +184,30 @@ public class VoiceRecorder {
                         break;
                     }
                     final int size = mAudioRecord.read(mBuffer, 0, mBuffer.length);
+                    final long now = System.currentTimeMillis();
                     if (isHearingVoice(mBuffer, size)) {
                         if (mLastVoiceHeardMillis == Long.MAX_VALUE) {
+                            mVoiceStartedMillis = now;
                             mCallback.onVoiceStart();
                         }
                         mCallback.onVoice(mBuffer, size);
-                        mLastVoiceHeardMillis = System.currentTimeMillis();
+                        mLastVoiceHeardMillis = now;
+                        if (now - mVoiceStartedMillis > MAX_SPEECH_LENGTH_MILLIS) {
+                            end();
+                        }
                     } else if (mLastVoiceHeardMillis != Long.MAX_VALUE) {
                         mCallback.onVoice(mBuffer, size);
-                        if (System.currentTimeMillis() - mLastVoiceHeardMillis
-                                > SPEECH_TIMEOUT_MILLIS) {
-                            mLastVoiceHeardMillis = Long.MAX_VALUE;
-                            mCallback.onVoiceEnd();
+                        if (now - mLastVoiceHeardMillis > SPEECH_TIMEOUT_MILLIS) {
+                            end();
                         }
                     }
                 }
             }
+        }
+
+        private void end() {
+            mLastVoiceHeardMillis = Long.MAX_VALUE;
+            mCallback.onVoiceEnd();
         }
 
         private boolean isHearingVoice(byte[] buffer, int size) {

--- a/speech/Speech/build.gradle
+++ b/speech/Speech/build.gradle
@@ -21,7 +21,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.3'
+        classpath 'com.android.tools.build:gradle:2.2.0'
         classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.0'
 
         // NOTE: Do not place your application dependencies here; they belong

